src/rgw/rgw_es_query.cc

```diff
@@ -149,15 +149,20 @@ class ESQueryNode_Bool : public ESQueryNode {
   }
   virtual ~ESQueryNode_Bool() {
     delete first;
-    delete second;
+    if (second) {
```

Comment:
not needed. `delete nullptr` is just no-op

Reply:
Yes, you are right